src/pyhf/patchset.py

```diff
@@ -0,0 +1,138 @@
+"""
+pyhf patchset provides a user-friendly interface for interacting with patchsets.
+"""
+import logging
+import jsonpatch
+from . import exceptions
+from . import utils
+
+log = logging.getLogger(__name__)
+
+
+class Patch(jsonpatch.JsonPatch):
+    """
+    A patch.
+    """
+
+    def __init__(self, spec):
+        """
+        Construct a Patch.
+
+        Args:
+            spec (`jsonable`): The patch JSON specification
+
+        Returns:
+            patch (`Patch`): The Patch instance.
+
+        """
+        super(Patch, self).__init__(spec['patch'])
+        self.metadata = spec['metadata']
+
+    def __repr__(self):
+        """ Representation of the patch object """
+        return f"<Patch object '{self.name}{self.values}' at {hex(id(self))}>"
+
```

Comment:
@kratsg  Maybe something like
```suggestion

    def __eq__(self, other):
        """ Equality for subclass with new attributes """
        if not isinstance(other, Patch):
            return False
        return Patch.__eq__(self, other) and self.metadata == other.metadata
```

to deal with https://lgtm.com/rules/9990086/ ?